### DISCUSSION TITLE
Understanding "areas" in Container, Context and List blocks.

### DISCUSSION THREAD

**Discussion by vaddisrinivas** [lowdefy/lowdefy#604] [General]
Sharing my understanding about the Block/Container/Context areas.

- Firstly, from the documentation, `All user interfaces in Lowdefy are assembled out of blocks`

- Every block/container has areas that can be defined.
-  One can find these `areas` at the very top of every block in the documentation - right under the title of the block.
  - For ex-
    - [Card ](https://docs.lowdefy.com/Card)
      - areas: content, extra, title
    - [Badge](https://docs.lowdefy.com/Badge) 
      - areas: content
    - [Modal](https://docs.lowdefy.com/Modal)
      - areas: content, footer
-  To customize the blocks areas could be defined/extended
-  by default (wherever applicable) the `blocks` section is mapped to `areas -> content`
________________________________________________________________________________________________________________________________________
Example:
The title of a `Card` could be defined simply by adding  the `title` property to the card block-
```        type: Card
        properties:
          title: This is the title
```

However, in order to customize the `title`  furthermore, the `title` could be defined as a separate section in the `areas` because the Card type allows `areas: content, extra, title`. So, it may be added as -

```      - id: test
        type: Card
        properties:
          headerStyle:
            marginTop: 0
            marginBottom: 0
            background: red
        areas:
          title:
            blocks:
              - id: one
                type: Html
                style:
                  fontSize: 10
                  md:
                    fontSize: 30
                properties:
                  html: <div>This is a title</div>
          content:
            blocks:
              ...
```
Advantage of the second approach is that it allows much deeper customization. We could add a `Box` or a block of type `Title` e.t.c.

Please note that the styles could however depend on how Ant/Lowdefy consider/render/apply them.

@Gervwyk  please do add your comments/corrections if needed

**Comment by Gervwyk** [comment]
Thanks, @vaddisrinivas  for the summery. Also, see the [Layout](https://docs.lowdefy.com/layout) page for more details and props regarding content areas. 

Also worth noting that only `container`, `context` and `list` block types can have content areas.
